### PR TITLE
Pmt vectors

### PR DESCRIPTION
**Pull request by jsallay**
Add in support for vectors of pmts.
Addresses https://github.com/gnuradio/pmt/issues/6

### CODE REVIEW

**Comment by jsallay** [comment]
I am adding in unit tests and I think that this is working out to be pretty cool.
I can declare a vector like this.
```
vector<pmt> x{5, "abc", std::complex<float>(1.0, -2.0), scalar<int>(3), pmt("z")};
```

File: include/pmtf/wrap.hpp [hunk 1]
```diff
@@ -112,44 +118,46 @@ inline std::ostream& operator<<(std::ostream& os, const map& value) {
...
+        case Data::ScalarComplex128: return operator<<(os, scalar<std::complex<double>>(value));
+        case Data::ScalarInt8: return operator<<(os, scalar<int8_t>(value));
+        case Data::ScalarInt16: return operator<<(os, scalar<int16_t>(value));
+        case Data::ScalarInt32: return operator<<(os, scalar<int32_t>(value));
+        case Data::ScalarInt64: return operator<<(os, scalar<int64_t>(value));
+        case Data::ScalarUInt8: return operator<<(os, scalar<uint8_t>(value));
+        case Data::ScalarUInt16: return operator<<(os, scalar<uint16_t>(value));
+        case Data::ScalarUInt32: return operator<<(os, scalar<uint32_t>(value));
+        case Data::ScalarUInt64: return operator<<(os, scalar<uint64_t>(value));
+        //case Data::ScalarBool: return operator<<(os, scalar<bool>(value));
```
Comment (mormj) [inline include/pmtf/wrap.hpp:138]: Why no bool?

File: include/pmtf/wrap.hpp [hunk 2]
```diff
@@ -20,10 +20,15 @@ namespace pmtf {
 inline std::ostream& operator<<(std::ostream& os, const map& value);
 bool operator==(const pmt& x, const map& y);
 inline bool operator==(const map& y, const pmt& x) { return operator==(x,y); }
-std::ostream& operator<<(std::ostream& os, const pmt& value);
```
Comment (mormj) [inline include/pmtf/wrap.hpp:23]: What is the general purpose of `wrap` at this point since most everything that was wrap got moved into the base class

**Comment by mormj** [comment]
Tested it out with newsched and a few small tweaks that I'll put in another PR.  Merging

**Comment by jsallay** [comment]
I need to open an issue for that.

I don't believe that bool will pass any is_arithmetic checks, so I would
just need to make sure that it works properly with everything else.

**Comment by jsallay** [comment]
Wrap really exists now as a run around the resolution order of the compiler.

I have to declare a lot of the functions like operator== after all of the
classes have been defined so that I can cast the pmt to those classes.

The file should be renamed to something that better represents this. I also
think that we need a pmt.h that collects all of the other headers. That way
a user doesn't need to know all of the details.

**Pull request merged**